src/main/kotlin/calculator/domain/Number.kt

```diff
@@ -0,0 +1,41 @@
+package calculator.domain
+
+class Number(val value: Int) {
+
+    constructor(value: String) : this(value.toIntOrNull() ?: throw IllegalArgumentException("숫자가 아님"))
+
+    fun plus(number: Number): Number {
+        return Number(this.value + number.value)
+    }
```

Comment:
operator overloading을 사용해보시면 어떨까요?

```suggestion
    operator fun plus(number: Number): Number {
        return Number(this.value + number.value)
    }
```

operator overloading을 사용하면 enum의 로직을 아래와 같이 변경할 수 있겠네요.

```kotlin
PLUS("+", { x, y -> x +y }),
```

https://kotlinlang.org/docs/operator-overloading.html#arithmetic-operators

Reply:
반영하였습니다. 89907b2
코틀린은 신기한게 많네요. 연산자를 함수처럼 사용할 수 있다니 😳